File: src/main/cs/Problem0003.cs
using System;
using System.Diagnostics;
using Euler.test.cs;

namespace Euler.main.cs
{
    public class Problem0003
    {
        public Problem0003()
        {
            Stopwatch stopWatch = new Stopwatch();
            stopWatch.Start();
            ulong result = 0;

            // Solution:
            ulong i;
            ulong maxNumber = 600_851_475_143;

            for (i = 2; i <= maxNumber; i++) {
                if (maxNumber % i == 0) {
                    maxNumber /= i;
                    i--;
                }
            }
            result = i;


            stopWatch.Stop();
            string elapsedTime = stopWatch.Elapsed.ToString();
            Console.WriteLine("Result:\t" + result + "\tTime:\t" +
                              (double.Parse(elapsedTime.Substring(elapsedTime.LastIndexOf(":") + 1, 2)) >= 1
                                  ? double.Parse(elapsedTime.Substring(elapsedTime.LastIndexOf(":") + 1)) + " s"
                                  : double.Parse(elapsedTime.Substring(elapsedTime.IndexOf(".") + 1)) / 10_000 +
                                    " ms"));
            if (Test.DoBenchmark)
            {
                Benchmark.AddTime(3, double.Parse(elapsedTime.Substring(elapsedTime.IndexOf(".") + 1)) / 10_000);
            }
        }
    }
}

File: src/main/java/de/swerik/euler/_00/Problem0003.java
package de.swerik.euler._00;

public class Problem0003 {
    public Problem0003() {
        long startTime = System.nanoTime();
        long result = 0;
//        for (int i = (int) Math.sqrt(600851475143L); i > 2; i--) {
//            if (600851475143L % i != 0) {
//                continue;
//            }
//            for (int j = 2; j < Math.sqrt(i); j++) {
//                if (i % j == 0) {
//                    break;
//                }
//                if (j == (int) Math.sqrt(i)) {
//                    result = i;
//                    break;
//                }
//            }
//            if (result != 0) {
//                break;
//            }
//        }

        int i;
        long maxNumber = 600851475143L;

        for (i = 2; i <= maxNumber; i++) {
            if (maxNumber % i == 0) {
                maxNumber /= i;
                i--;
            }
        }
        result = i;

        long timeToResolve = System.nanoTime() - startTime;
        if (((double) timeToResolve / 1_000_000) >= 1000) {
            System.out.println("Result:\t" + result + "\tTime:\t" + ((double) timeToResolve / 1_000_000_000) + "s");
        } else {
            System.out.println("Result:\t" + result + "\tTime:\t" + ((double) timeToResolve / 1_000_000) + "ms");
        }
    }
}
